### ISSUE TITLE
Throw exception when a Class invoking InetAddress.getLocalHost() in static scope

### ISSUE THREAD

**Issue by rim99** [oracle/helidon#2483]
## Environment Details
* Helidon Version: 2.0.2
* Helidon SE
* JDK version: JDK11
* OS: MacOSX 10.15.7
* Docker version (if applicable): Docker version 19.03.12, build 48a66213fe

----------

## Problem Description

If a Class is invoking `InetAddress.getLocalHost()` method in static scope.

When running the app inside a docker container, the static field will hold a strange string other than "localhost" or any other hostname we usually see.

The stack trace is 

```
Creating transaction-consumer_hostname-demo_1 ... done
Attaching to transaction-consumer_hostname-demo_1
hostname-demo_1     | java.net.UnknownHostException: ffc714a90c26: ffc714a90c26: Name or service not known
hostname-demo_1     | 	at java.net.InetAddress.getLocalHost(InetAddress.java:1642)
hostname-demo_1     | 	at org.example.SomeClass.<clinit>(SomeClass.java:11)
hostname-demo_1     | 	at com.oracle.svm.core.hub.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:350)
hostname-demo_1     | 	at com.oracle.svm.core.hub.ClassInitializationInfo.initialize(ClassInitializationInfo.java:270)
hostname-demo_1     | 	at java.lang.Class.ensureInitialized(DynamicHub.java:499)
hostname-demo_1     | 	at org.example.Demo.main(Demo.java:9)
hostname-demo_1     | Caused by: java.net.UnknownHostException: ffc714a90c26: Name or service not known
hostname-demo_1     | 	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_net_UnknownHostException_2_0002e_0003cinit_0003e_00028Ljava_lang_String_2_00029V(JNIJavaCallWrappers.java:0)
hostname-demo_1     | 	at java.net.Inet4AddressImpl.lookupAllHostAddr(Inet4AddressImpl.java)
hostname-demo_1     | 	at java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
hostname-demo_1     | 	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)
hostname-demo_1     | 	at java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
hostname-demo_1     | 	at java.net.InetAddress.getAllByName0(InetAddress.java:1505)
hostname-demo_1     | 	at java.net.InetAddress.getLocalHost(InetAddress.java:1637)
hostname-demo_1     | 	... 5 more
hostname-demo_1     | Exception in thread "main" java.net.UnknownHostException: ffc714a90c26: ffc714a90c26: Name or service not known
hostname-demo_1     | 	at java.net.InetAddress.getLocalHost(InetAddress.java:1642)
hostname-demo_1     | 	at org.example.Demo.main(Demo.java:10)
hostname-demo_1     | Caused by: java.net.UnknownHostException: ffc714a90c26: Name or service not known
hostname-demo_1     | 	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_net_UnknownHostException_2_0002e_0003cinit_0003e_00028Ljava_lang_String_2_00029V(JNIJavaCallWrappers.java:0)
hostname-demo_1     | 	at java.net.Inet4AddressImpl.lookupAllHostAddr(Inet4AddressImpl.java)
hostname-demo_1     | 	at java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
hostname-demo_1     | 	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1515)
hostname-demo_1     | 	at java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
hostname-demo_1     | 	at java.net.InetAddress.getAllByName0(InetAddress.java:1505)
hostname-demo_1     | 	at java.net.InetAddress.getLocalHost(InetAddress.java:1637)
hostname-demo_1     | 	... 1 more
transaction-consumer_hostname-demo_1 exited with code 1
```

## Steps to reproduce

I create a demo can reproduce this problem: https://github.com/rim99/hostname-helidon-se-demo

Try run the following commands, and compare the output.

```
docker-compose up demo-helidon
docker-command up demo-graal
```

**Comment by romain-grecourt** [comment]
This seems to be related to https://github.com/oracle/graal/issues/1338 which is fixed in GraalVM 20.2.0.

`Dockerfile.graal` uses `oracle/graalvm-ce:20.2.0-java11` and `Dockerfile.helidon` uses `helidon/jdk11-graalvm-maven:20.1.0` ; this is the reason why demo-graal works but demo-helidon does not.

If you change `Dockerfile.helidon` to use `oracle/graalvm-ce:20.2.0-java11` it's working fine. See
```diff
diff --git a/Dockerfile.helidon b/Dockerfile.helidon
index b372848..d704ff6 100644
--- a/Dockerfile.helidon
+++ b/Dockerfile.helidon
@@ -13,16 +13,12 @@ RUN mvn package -Pnative-image -Dnative.image.skip -Dmaven.test.skip -Declipseli
 # Do the Maven build!
 # Incremental docker builds will resume here when you change sources
 ADD src src
-RUN mvn package -Pnative-image -Dnative.image.buildStatic -DskipTests
-
-# 2nd stage, build the runtime image
-FROM scratch
-# FROM debian:bullseye-20201012-slim
-WORKDIR /helidon
-
-# Copy the binary built in the 1st stage
-COPY --from=build /helidon/target/hostname-demo .
-
+RUN mvn package -DskipTests
+RUN ls -l target/
+
+FROM oracle/graalvm-ce:20.2.0-java11 as compile
+WORKDIR /app
+RUN gu install native-image
+COPY --from=build /helidon/target/hostname-demo.jar ./hostname-demo.jar
+RUN native-image -jar hostname-demo.jar
 ENTRYPOINT ["./hostname-demo"]
-
-EXPOSE 8080
```

We will provide a new version of `helidon/jdk11-graalvm-maven` as part of https://github.com/oracle/helidon/issues/2477 ; 20.2.0 or later.

**Comment by rim99** [comment]
Thank you for the help 👍
